src/reducer.ts

```diff
@@ -78,6 +78,17 @@ export default (oldState: State, action: Action): State => {
   const state: State = JSON.parse(JSON.stringify(oldState))
   state.prepopulatedInput = undefined
 
+  // Only join local users into the userMap if we don't already have them - server users are fresher & may contain updates
+  const localUserMapString = localStorage.getItem('userMap')
```

Comment:
We probably don't want to do this every reducer run, only on pageload, yeah?

My assumption was this would happen at the same time we load messages from localStorage. We could even use the same timestamp check, since if we don't have any old messages floating around we don't have use for old user map data.